components/datetime/src/format.rs

```diff
@@ -0,0 +1,56 @@
+use super::date::DateTime;
+// use super::pattern::{parse_pattern, FieldType};
+use crate::fields::{FieldLength, FieldSymbol};
+use crate::pattern::{Pattern, PatternItem};
+use std::fmt;
+
+pub struct FormattedDateTime<'l> {
+    pub(crate) pattern: &'l Pattern<'static>,
+    pub(crate) display_names: &'l crate::provider::DisplayNames<'static>,
+    pub(crate) date_time: &'l DateTime,
+}
+
+impl<'l> fmt::Display for FormattedDateTime<'l> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_pattern(self.pattern, self.display_names, &self.date_time, f)
+    }
+}
+
+fn format_number(
+    result: &mut impl fmt::Write,
+    num: usize,
+    length: &FieldLength,
+) -> Result<(), std::fmt::Error> {
+    write!(result, "{:0>width$}", num, width = (*length as u8) as usize)
```

Comment:
I don't understand the `(*length as u8) as usize`.  FieldLength does not correspond to the number of zeros of numeric fields.

Reply:
It does.

Reply:
The definition is:

```
#[derive(Debug, PartialEq, Clone, Copy)]
pub enum FieldLength {
    One = 1,
    TwoDigit = 2,
    Abbreviated = 3,
    Wide = 4,
    Narrow = 5,
    Six = 6,
}
```

Note that Narrow and Six do not correspond to the number of zeros.  One and TwoDigit probably do.

Reply:
Please write a comment explaining `(*length as u8) as usize` and then resolve.